File: ruby/models/customer_success.rb
require_relative 'customer'

class CustomerSuccess
  attr_accessor :id, :score, :customers

  def initialize(id: 0, score: 0)
    @customers = []
    @id = id
    @score = score
  end

  def add_customer(customer)
    customer.add_customer_success(self)
    @customers << customer
  end

  def size_of_customers
    @customers.size
  end
end

File: javascript/models/customer-success.js
class CustomerSuccess {
    constructor({ id, score }) {
     this.customers = [];
     this.id = id;
     this.score = score;
    }

    addCustomer = (customer) => {
      this.customers.push(customer);
      customer.addCustomerSuccess(this);
    }
}

module.exports = { CustomerSuccess };